cmd/insights/contributors.go

```diff
@@ -0,0 +1,271 @@
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"github.com/charmbracelet/bubbles/table"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/open-sauced/go-api/client"
+	"github.com/open-sauced/pizza-cli/pkg/api"
+	"github.com/open-sauced/pizza-cli/pkg/constants"
+	"github.com/open-sauced/pizza-cli/pkg/utils"
+	"github.com/spf13/cobra"
+)
+
+type Options struct {
+	// APIClient is the http client for making calls to the open-sauced api
+	APIClient *client.APIClient
+
+	// Repos is the array of git repository urls
+	Repos []string
+
+	// FilePath is the path to yaml file containing an array of git repository urls
+	FilePath string
+
+	// Period is the number of days, used for query filtering
+	Period int32
+}
+
+// NewContributorsCommand returns a new cobra command for 'pizza insights contributors'
+func NewContributorsCommand() *cobra.Command {
+	opts := &Options{}
+	cmd := &cobra.Command{
+		Use:   "contributors [flags]",
+		Short: "Gather insights about contributors of indexed git repositories",
+		Long:  "Gather insights about contributors of indexed git repositories. This command will show new, recent, alumni, repeat contributors for each git repository",
+		Args: func(cmd *cobra.Command, args []string) error {
+			fileFlag := cmd.Flags().Lookup(constants.FlagNameFile)
+			if !fileFlag.Changed && len(args) == 0 {
+				return fmt.Errorf("must specify git repository url argument(s) or provide %s flag", fileFlag.Name)
+			}
+			opts.Repos = append(opts.Repos, args...)
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			endpointURL, _ := cmd.Flags().GetString(constants.FlagNameEndpoint)
+			opts.APIClient = api.NewGoClient(endpointURL)
+			return run(opts)
+		},
+	}
+	cmd.Flags().StringVarP(&opts.FilePath, constants.FlagNameFile, "f", "", "Path to yaml file containing an array of git repository urls")
+	cmd.Flags().Int32VarP(&opts.Period, constants.FlagNamePeriod, "p", 30, "Number of days, used for query filtering")
+	return cmd
+}
+
+func run(opts *Options) error {
+	repositories, err := utils.HandleRepositoryValues(opts.Repos, opts.FilePath)
+	if err != nil {
+		return err
+	}
+	var (
+		waitGroup = new(sync.WaitGroup)
+		errorChan = make(chan error, len(repositories))
+	)
+	for url := range repositories {
+		repoURL := url
+		waitGroup.Add(1)
+		go func() {
+			defer waitGroup.Done()
+			repoContributorsInsights, err := findAllRepositoryContributorsInsights(context.TODO(), opts, repoURL)
+			if err != nil {
+				errorChan <- err
+				return
+			}
+			repoContributorsInsights.RenderTable()
+		}()
+	}
+	waitGroup.Wait()
+	close(errorChan)
+	var allErrors error
+	for err = range errorChan {
+		allErrors = errors.Join(allErrors, err)
+	}
+	return allErrors
+}
+
+type repositoryContributorsInsights struct {
+	RepoID  int32
+	RepoURL string
+	New     []string
+	Recent  []string
+	Alumni  []string
+	Repeat  []string
+}
+
+func (rci *repositoryContributorsInsights) RenderTable() {
```

Comment:
Generally speaking, and this can be an additional issue we work on after this merges, we should also always provide a `-o / --output` flag that can be either "yaml" or "json" since we want people to be able to compose their data into other tools (like jq, grep, sed, awk, etc.)

Reply:
I understand, I can refactor this to allow such an option as part of the work for https://github.com/open-sauced/pizza-cli/issues/10 👍🏻